lib/RWMutex.ts

```diff
@@ -227,21 +234,47 @@ export default class RWMutex {
    */
   async conditionalOverrideLockWriter(
     conditional: (oldWriter: string, newWriter: string) => Promise<boolean>,
-    upsert = false): Promise<boolean> { 
-    const mongoLock = await this._coll.findOne({ lockID: this._lockID })
-    if (!mongoLock) {
-      if (!upsert) {
-        return false;
+    upsert = true, timeout = 10000): Promise<boolean> { 
+    const start = Date.now();
+    
+    while (Date.now() - start < timeout) {
+      const mongoLock = await this._coll.findOne({ lockID: this._lockID });
+      if (!mongoLock) {
+        if (!upsert) {
+          return false;
+        }
+        try {
+          await this.tryOverrideLockWriter("", upsert);
```

Comment:
I'm a little lost on what this try clause is doing. Can you explain it to me like I'm 5?

Reply:
this try clause is saying if upsert is true and there is no lock matching our query we can attempt an override.  The override will just create the lock if it doesn't exist yet.  The empty string here is just because we don't expect an old lock to exist with writer value to match against

Reply:
gotcha! This is the part I was missing "The empty string here is just because we don't expect an old lock to exist with writer value to match against". Thanks!